src/charm.py

```diff
@@ -1026,17 +995,21 @@ def render_pgb_config(self, reload_pgbouncer=False, restart=False) -> None:
         most convenient to have a single source of truth for the whole config.
 
         Args:
-            reload_pgbouncer: A boolean defining whether or not to reload the pgbouncer application
-                in the container. When config files are updated, pgbouncer must be restarted for
-                the changes to take effect. However, these config updates can be done in batches,
-                minimising the amount of necessary restarts.
             restart: Whether to restart the service when reloading.
         """
         perm = 0o400
 
-        if not self.configuration_check():
+        pgb_container = self.unit.get_container(PGB)
+        if not self.configuration_check() or not pgb_container.can_connect():
             return
 
+        # Transient file
+        auth_file = f"/dev/shm/{self.app.name}_{pgb.generate_password()}"  # noqa: S108
+        userlist = self.get_secret(APP_SCOPE, AUTH_FILE_DATABAG_KEY)
+        if not userlist:
+            userlist = ""
+        auth_type = "md5" if f'"{self.backend.stats_user}" "md5' in userlist else "scram-sha-256"
```

Comment:
Old monitoring users will fail to connect during upgrade, so using md5 auth until the user list is updated. New installs should be using SCRAM from the start.